include/zio/zio_dev.h

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright (c) 2019 Thomas Burdick <thomas.burdick@gmail.com>
+ * Copyright (c) 2019 Kevin Townsend
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief ZIO Device, a pollable sampled IO device
+ *
+ * ZIO devices are composed of device attributes (zio_dev_attr),
+ * channels (zio_dev_chan) and channel attributes (zio_chan_attr) in the
+ * following manner:
+ *
+ * ZIO Device----------------------------------------------------------------+
```

Comment:
If you add five spaces after the ``*`` (four spaces relative to the previous text line) this diagram will be rendered in the API HTML docs as a verbatim text block.  (Otherwise it will all be rendered as a normal HTML paragraph and won't come out looking as you intended:

```
 *     ZIO Device----------------------------------------------------------------+
 *     |                                                                         |
```